components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -291,9 +289,12 @@ ConversationHandler* AIChatService::CreateConversationHandlerForContent(
     int associated_content_id,
     base::WeakPtr<AssociatedContentDelegate> associated_content) {
   ConversationHandler* conversation = CreateConversation();
-  // Provide the content delegate, if allowed
-  MaybeAssociateContent(conversation, associated_content_id,
-                        associated_content);
+  // Provide the content delegate, if allowed. If we aren't initially enabling
+  // the context we still need to call MaybeAssociateContent so the conversation
+  // knows what the current tab is.
+  MaybeAssociateContent(
+      conversation, associated_content_id,
+      features::IsPageContextEnabledInitially() ? associated_content : nullptr);
```

Comment:
Now the UI doesn't explicitly control whether we send the contents or not we want to
1. Associate the tab with the conversation
2. Not actually attach it.

This will be more obvious once https://github.com/brave/brave-core/pull/29680 lands and completely removes the |should_send| flag and explicitly adds/removes the contents.